services/src/main/java/com/chatbot/services/MessageSenders/MessageSender.java

```diff
@@ -0,0 +1,8 @@
+package com.chatbot.services.MessageSenders;
+
+import java.io.IOException;
+
+abstract class MessageSender {
+    abstract void sendMessage(final String chatClientGeneratedID, final String msg)
```

Comment:
I feel you could rename sendMessageBasedOnMessageType. could have three args: ID, msg, msg_type. msg_type could be an enum {TEXT=1,CARD=2,CARD_WITH_TEXT_HEADER=3}.

Impl could be like:
sendMessageBasedOnMessageType () {
List<Message> m = GenerateMessageListByType(string msg, msgtype);
m.stream().map(m -> SendMessage(id,m));
}
GenerateMessageListByType(string msg, msgtype) {
List<Message> m;
switch (msgtype) {
case TEXT:
m.add(generateTextMessge()); break;
case CARD:
m.add(generateCardMessage());break;
case CARD_WITH_HEADER:
m.add(generateTextMessge());
m.add(generateCardMessage()); break;
}
return m;
}


}
}

Reply:
updated